DFC.TestAutomation.UI/TestSupport/Configurator.cs

```diff
@@ -6,55 +6,78 @@
 
 namespace DFC.TestAutomation.UI.TestSupport
 {
-    public static class Configurator
+    public  class Configurator
     {
-        private static IConfigurationRoot _config;
+        private  IConfigurationBuilder _builder;
+        
+        private  IConfigurationRoot _config;
 
-        private static IConfigurationRoot _hostingConfig;
+        private  IConfigurationRoot _hostingConfig;
 
-        public static bool IsVstsExecution { get; set; }
+        public  bool IsVstsExecution { get; set; }
 
-        public static string EnvironmentName { get; set; }
+        public  string EnvironmentName { get; set; }
 
-        public static string ProjectName { get; set; }
+        public  string ProjectName { get; set; }
 
-        public static IConfigurationRoot GetConfig() => _config;
+        public Configurator()
+        {
+            _builder = ConfigurationBuilder();
+        }
+
+        public  IConfigurationRoot GetConfig() => _config;
+
+        /// <summary>
+        /// Add an appsetting json file to configuration before build it
+        /// </summary>
+        /// <param name="fileName">Name of an appsettings file that is located in the project (relative to the base path) eg "appsettings.json"</param>
+        /// <returns></returns>
+        public Configurator AddSettingsFile(string fileName)
+        {
+            _builder.AddJsonFile(fileName, true);
+            return this;
+        }
 
-        public static IConfigurationRoot InitializeConfig(string[] settingsFiles)
+        /// <summary>
```

Comment:
Id be wary of adding summaries to the methods at this stage. Theyre likely to change

Reply:
fair enough. I mainly just wanted to see if it will result in hints in the IDE when using the Nuget